### PR TITLE
add tests for distribution generators

### PR DESCRIPTION
**Pull request by sxjscience**
## Description ##
Add the tests required in testing https://github.com/apache/incubator-mxnet/pull/9119

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Add mean_check, var_check, chi_square_check APIs for testing random generators
- [x] Add tests for all the current supported random generators.

### CODE REVIEW

File: tests/python/unittest/test_random.py [hunk 1]
```diff
@@ -216,6 +220,91 @@ def test_sample_multinomial():
...
+
+def test_multinomial_generator():
+    print('Multinomial Distribution Test:')
+    ctx = mx.context.current_context()
+    probs = [0.1, 0.2, 0.3, 0.05, 0.15, 0.2]
+    sample_num = 1000000
+    buckets = list(range(6))
+    for dtype in ['float16', 'float32', 'float64']:
+        print("ctx=%s, dtype=%s" %(ctx, dtype))
+        generator_mx = lambda x: mx.nd.random.multinomial(data=mx.nd.array(np.array(probs), ctx=ctx, type=dtype),
```
Comment (javelinjs) [inline tests/python/unittest/test_random.py]: should be `dtype=dtype`?

Comment (sxjscience) [reply tests/python/unittest/test_random.py]: Yes, should be dtype

**Comment by marcoabreu** [comment]
Please consider removing the print-statements as nose will handle outputting the names of all tests and various debug infos around it.

**Comment by sxjscience** [comment]
@marcoabreu I've removed the unnecessary prints. Need @javelinjs to merge this in.

**Comment by javelinjs** [comment]
changes merged to https://github.com/apache/incubator-mxnet/pull/9119

**Comment by sxjscience** [comment]
@marcoabreu I'm following the implementation of ndarray.sparse (https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/ndarray/sparse.py#L58-L61). However, I should raise a warning if ss is not found.

**Comment by marcoabreu** [comment]
For tests, please throw an error instead of a warning if a dependency is not present.

**Comment by sxjscience** [comment]
@marcoabreu Yes, should be an error.

**Comment by marcoabreu** [comment]
Okay, thank you. I don't think scipi is installed inside our test environments so your PR may fail until we added a proper dependency management for CI. I will notify you as soon as this has been added.

**Comment by javelinjs** [comment]
Close since https://github.com/apache/incubator-mxnet/pull/9119 is merged. Thanks @sxjscience !

**Pull request closed without merging**